base/array.jl

```diff
@@ -1061,13 +1051,13 @@ julia> findnext(isodd, A, 2)
 0
 ```
 """
-function findnext(testf::Function, A, start::Integer)
-    for i = start:length(A)
-        if testf(A[i])
-            return i
-        end
+function findnext(testf::Function, A, idx::Integer)
+    while !done(A, idx)
```

Comment:
Probably, but that would be yet another function to implement for custom types. I guess we could have it fall back to `identity` so that only iterators which deviate from the standard need to implement it.